### PR TITLE
Maven plugin upgrade 3.5.4

### PR DESCRIPTION
**Pull request by duttonw**
3.1.1 to 3.5.4 maven plugin upgrade

### CODE REVIEW

**Comment by duttonw** [comment]
Have included all of the dependabot pull requests.

**Comment by duttonw** [comment]
Fix conflicts with master and is ready to be integrated.

**Comment by duttonw** [comment]
Now included https://github.com/vorburger/MariaDB4j/pull/176

**Comment by vorburger** [comment]
@duttonw FYI #207 ... I somehow thought that we had agreed in another issue that you only wanted to merge Maven upgrade related things after I cut release? I would like to release now and hit #207 ... but note #208 / #209.  In hindsight I now realize that this had important fixes needed before releasing.. but I'll go ahead with my #210 - will you rebase this after?  -- Sorry for the confusion on my end!

**Comment by vorburger** [comment]
@duttonw I don't know if you still have any interest in this project, and would be willing to resolve the merge conflicts which this PR meanwhile has? If you do - I'd love to merge it!

**Comment by duttonw** [comment]
I'll see what I can do.

**Comment by duttonw** [comment]
travis ci is happy.
https://travis-ci.org/vorburger/MariaDB4j/builds/579587853?utm_source=github_status&utm_medium=notification
Please review the changes as its been a while since I did the work.

My work currently does not use this plugin at moment, hopefully linux-ized container systems become easier to run (i.e. not docker with its security root privileges requirements) so we can streamline this even more and maybe include postgres as an optional db. (Since i've not found may like that)

File: mariaDB4j-maven-plugin/src/it/mariadb4j-maven-plugin-test-basic/pom.xml [hunk 1]
```diff
@@ -77,6 +83,7 @@
         <dependency>
             <groupId>mysql</groupId>
             <artifactId>mysql-connector-java</artifactId>
+            <version>8.0.12</version>
```
Comment (vorburger) [inline mariaDB4j-maven-plugin/src/it/mariadb4j-maven-plugin-test-basic/pom.xml]: we're using `8.0.13` in `mariaDB4j-maven-plugin/pom.xml`, but oh well - doesn't matter.

File: mariaDB4j-maven-plugin/src/test/java/ch/vorburger/mariaDB4j/Mariadb4jStartMojoTest.java [hunk 2]
```diff
@@ -41,9 +41,9 @@
 import java.util.Map;
 import java.util.Set;
 
-import static com.google.common.base.Preconditions.checkNotNull;
-import static com.google.common.base.Preconditions.checkState;
-import static org.junit.Assert.*;
+import static org.assertj.core.api.Assertions.assertThat;
```
Comment (vorburger) [inline mariaDB4j-maven-plugin/src/test/java/ch/vorburger/mariaDB4j/Mariadb4jStartMojoTest.java]: FYI personally I now actually prefer Google Truth https://truth.dev over AssertJ - but doesn't realy matter.

File: mariaDB4j-maven-plugin/pom.xml [hunk 3]
```diff
@@ -115,41 +133,40 @@
...
                 <artifactId>maven-plugin-testing-harness</artifactId>
-                <version>3.1.0</version>
-                <scope>test</scope>
+                <version>3.3.0</version>
             </dependency>
             <dependency>
+                <scope>test</scope>
                 <groupId>org.apache.maven</groupId>
                 <artifactId>maven-compat</artifactId>
                 <version>3.6.1</version>
```
Comment (vorburger) [inline mariaDB4j-maven-plugin/pom.xml]: @duttonw FYI we're actually already at 3.6.1 not just 3.5.4 here, but I'm sure that's fine...

**Comment by vorburger** [comment]
FTR: I usually always "Rebase and merge", but for this PR that would require more work to resolve conflics, so I've (exceptionally) used a Merge commit to get this one in - no harm done (I just find merge commits make the history harder to understand, so prefer rebase merging, when possible).

**Comment by duttonw** [comment]
Unsure, from subversion days, merge's are a part of life. I do agree on squash commits on feature branches, rebases also nice if can be done.

**Comment by vorburger** [comment]
> hopefully linux-ized container systems become easier to run (i.e. not docker with its security root privileges requirements) 

==> https://podman.io ! 

> so we can streamline this even more 

FYI https://www.testcontainers.org/modules/databases/mariadb/ (and https://www.testcontainers.org/modules/databases/mysql/) ... I had even mentioned it on https://github.com/vorburger/MariaDB4j#faq - probably too hidden?  Have also just created new #296 FYI.

> and maybe include postgres as an optional db. (Since i've not found may like that)

FYI https://www.testcontainers.org/modules/databases/postgres/

**Comment by duttonw** [comment]
Have been following podman, sadly not windows ready yet :'( https://github.com/containers/libpod/issues/1384
Would love to be able to wrap a docker system inside maven like i've been able to do with nodejs. reduce the infection rate of needing to system install so many packages. i.e [frontend-maven-plugin](https://github.com/eirslett/frontend-maven-plugin)

tho stating that, have been introduced to https://github.com/ankane/ahoy which is in use for building docker-ised containers running python for a ckan website with cucumber testing... i.e. [build.sh](https://github.com/qld-gov-au/ckanext-data-qld-theme/blob/develop/.circleci/build.sh ) [.ahoy.yml](https://github.com/qld-gov-au/ckanext-data-qld-theme/blob/develop/.ahoy.yml)

such a weird world it has become ;)

File: mariaDB4j-maven-plugin/pom.xml [hunk 4]
```diff
@@ -8,72 +8,90 @@
...
         </dependency>
     </dependencies>
     <dependencyManagement>
         <dependencies>
+            <dependency>
+                <groupId>${project.groupId}</groupId>
+                <artifactId>mariaDB4j</artifactId>
+                <version>${project.version}</version>
+            </dependency>
+            <!-- Maven testing includes an older version at a higher level tham mariaDB4j, so including here -->
```
Comment (sherrif10) [inline mariaDB4j-maven-plugin/pom.xml:89]: This should be "than" not "tham" when commenting, typo error

Comment (vorburger) [reply mariaDB4j-maven-plugin/pom.xml:89]: You are welcome to contribute a pull request to fix this typo.

Comment (sherrif10) [reply mariaDB4j-maven-plugin/pom.xml:89]: @vorburger thanks let me fix that now

**Comment by sherrif10** [comment]
Following commit [here ](https://github.com/vorburger/MariaDB4j/pull/361) cc @vorburger

**Pull request merged**